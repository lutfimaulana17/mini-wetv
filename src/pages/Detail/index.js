import React, { useState, useEffect } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { ICStar, ICLike } from '../../assets';
import { Header, Footer } from '../../components';

import { clearMoviePopular, requestMoviePopular, clearMovieDetail, requestMovieDetail } from '../../redux/actions/Movie';
import moment from 'moment';

const Detail = (props) => {
    const dispatch = useDispatch()
    const stateGlobal = useSelector(state => state)

    useEffect(() => {
        dispatch(requestMoviePopular())
        dispatch(requestMovieDetail(props.match.params.id))
        return () => {
            dispatch(clearMoviePopular())
            dispatch(clearMovieDetail())
        }
    }, [])

    const changeFormatDate = (date) => {
        return moment(date).format('LL')
    } 

    const { moviePopular, movieDetail } = stateGlobal.Movie;
    return (
        <div className="detail">
            <Header />
            <div className="detail-video">
            { movieDetail && Object.keys(movieDetail).length > 0 &&
                <img src={`https://image.tmdb.org/t/p/w1280${movieDetail.backdrop_path}`} style={{ height: '450px', width: '100%', objectFit: 'cover' }}/>
            }
            </div>
            <div className="detail-main">
            { movieDetail && Object.keys(movieDetail).length > 0 &&
                <div className="detail-main-left">
                    <div className="detail-main-left-header">
                        <img className="detail-main-left-header-img" src={`https://image.tmdb.org/t/p/w200${movieDetail.poster_path}`} />  
                        <div className="detail-main-left-header-content">
                            <p className="detail-main-left-header-content-title">{movieDetail.title}</p>
                            <p className="detail-main-left-header-content-desc">Release: {changeFormatDate(movieDetail.release_date)}</p>
                            <div className="detail-main-left-header-content-rating">
                                <img src={ICStar} className="detail-main-left-header-content-rating-img" />
                                <p className="detail-main-left-header-content-rating-text">{movieDetail.vote_average}</p>
                            </div>
                            <div className="detail-main-left-header-content-favorite">
                                <p className="detail-main-left-header-content-favorite-text">Set Favorite</p>
                                <img src={ICLike} className="detail-main-left-header-content-rating-img" />
                            </div>
                        </div>
                    </div>                
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title" id="detail-status">Status</p>
                        <p className="detail-main-left-item-desc">{movieDetail.status}</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Original Title</p>
                        <p className="detail-main-left-item-desc">{movieDetail.original_title}</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Genre</p>
                        <p className="detail-main-left-item-desc">{movieDetail.genres.map((item) => item.name).join(",")}</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Overview</p>
                        <p className="detail-main-left-item-desc">{movieDetail.overview}</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Production Companies</p>
                        <p className="detail-main-left-item-desc">{movieDetail.production_companies.map((item) => item.name).join(",")}</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Production Countries</p>
                        <p className="detail-main-left-item-desc">{movieDetail.production_countries.map((item) => item.name).join(",")}</p>
                    </div>
                </div>
            }
                <div className="detail-main-right">
                    <p className="detail-main-right-title">Film Terpopuler</p>
                    { moviePopular ?
                        moviePopular.results.length > 0 ?
                            moviePopular.results.slice(0, 4).map((item, index) =>
                                <Link to={`/detail/${item.id}`} style={{ textDecoration: 'none' }} key={index}>
                                    <div className="detail-main-right-item">
                                        <img src={`https://image.tmdb.org/t/p/w200${item.poster_path}`} className="detail-main-right-item-img" />
                                        <p className="detail-main-right-item-title">{item.title}</p>
                                    </div>
                                </Link>
                            )
                            : <p>No data!</p>
                        : <p>Loading...</p>
                    }
                </div>
            </div>
            <Footer />
        </div>
    )
}

export default withRouter(Detail)
