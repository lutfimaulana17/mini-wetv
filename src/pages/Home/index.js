import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from 'react-responsive-carousel';

import { Header, Footer, ItemMovie } from '../../components'
import { clearMovieLatest, clearMovieNowPlaying, clearMoviePopular, clearMovieTopRated, requestMovieLatest, requestMovieNowPlaying, requestMoviePopular, requestMovieTopRated, requestMovieUpcoming, clearMovieUpcoming } from '../../redux/actions/Movie';


const Home = () => {
    const dispatch = useDispatch()
    const stateGlobal = useSelector(state => state)

    const [dataMenu, setDataMenu] = useState([
        {
            id: 1,
            name: 'Semua'
        },
        {
            id: 2,
            name: 'Now Playing'
        },
        {
            id: 3,
            name: 'Popular'
        },
        {
            id: 4,
            name: 'Top Rated'
        },
        {
            id: 5,
            name: 'Upcoming'
        }
    ])
    const [menuActive, setMenuActive] = useState(1)

    useEffect(() => {
        dispatch(requestMovieLatest())
        dispatch(requestMovieNowPlaying())
        dispatch(requestMoviePopular())
        dispatch(requestMovieTopRated())
        dispatch(requestMovieUpcoming())
        return () => {
            dispatch(clearMovieLatest())
            dispatch(clearMovieNowPlaying())
            dispatch(clearMoviePopular())
            dispatch(clearMovieTopRated())
            dispatch(clearMovieUpcoming())
        }
    }, [])

    const { movieNowPlaying, moviePopular, movieTopRated, movieUpcoming } = stateGlobal.Movie;
    return (
        <div className="home">
            <Header />
            <div className="home-wrapper-caurosel">
                <Carousel autoPlay={true} interval={3000} showThumbs={false} showStatus={false}>
                    { moviePopular ?
                        moviePopular.results.length > 0 ?
                            moviePopular.results.slice(0, 5).map((item, index) => 
                                    <Link to={`/detail/${item.id}`} style={{ textDecoration: 'none' }} key={index}>
                                        <div key={item.id}>
                                            <img src={`https://image.tmdb.org/t/p/w1280${item.backdrop_path}`} style={{ height: '450px', objectFit: 'cover' }}/>
                                        </div>
                                    </Link>
                                )
                            : <p>No data!</p>
                        : <p>Loading...</p>
                    }
                </Carousel>
            </div>
            <div className="home-menu">
                { dataMenu.map((item, index) => 
                    <div id={`menu-film${item.id}`} className={item.id === menuActive ? "home-menu-item home-menu-item-active" : "home-menu-item"} key={index} onClick={() => setMenuActive(item.id)}>
                        <p className={item.id === menuActive ? "home-menu-item-text home-menu-item-text-active" : "home-menu-item-text"} >{item.name}</p>
                    </div>
                ) }
            </div>
            <div className="home-main">
                { (menuActive === 1 || menuActive === 2) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Now Playing</p>
                        <div className="home-main-section-content">
                            { movieNowPlaying ?
                                movieNowPlaying.results.length > 0 ?
                                    movieNowPlaying.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
                { (menuActive === 1 || menuActive === 3) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Popular</p>
                        <div className="home-main-section-content">
                            { moviePopular ?
                                moviePopular.results.length > 0 ?
                                    moviePopular.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
                { (menuActive === 1 || menuActive === 4) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Top Rated</p>
                        <div className="home-main-section-content">
                            { movieTopRated ?
                                movieTopRated.results.length > 0 ?
                                    movieTopRated.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
                { (menuActive === 1 || menuActive === 5) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Upcoming</p>
                        <div className="home-main-section-content">
                            { movieUpcoming ?
                                movieUpcoming.results.length > 0 ?
                                    movieUpcoming.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
            </div>
            <Footer />
        </div>
    )
}

export default Home
