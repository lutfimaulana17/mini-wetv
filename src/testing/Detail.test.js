import React from "react";
import { shallow, mount } from "enzyme";
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import toJson from "enzyme-to-json";
import moment from 'moment';

import { Detail } from "../pages";
import store from '../redux/store';

const changeFormatDate = (date) => {
    return moment(date).format('LL')
} 

describe("rendering components detail", () => {
    it("renders Detail component without crashing", () => {
        shallow(<Provider store={store}>
                    <Router>
                        <Detail />
                    </Router>
                </Provider>);
    })
})

describe("test function components detail", () => {
    const date = changeFormatDate('2021-08-25');
    expect(date).toEqual('August 25, 2021');
})

describe("snapshots component detail", () => {
    it("Detail snapshots", () => {
      const detailSnap = shallow(<Provider store={store}><Router><Detail /></Router></Provider>);
      expect(toJson(detailSnap)).toMatchSnapshot()
    })
})