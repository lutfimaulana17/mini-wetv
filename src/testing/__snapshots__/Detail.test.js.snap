// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots component detail Detail snapshots 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <withRouter(Detail) />
  </BrowserRouter>
</ContextProvider>
`;
